### PR TITLE
Wrap deletion operations in transactions

### PR DESCRIPTION
**Pull request by scottx611x**
Addresses #1580

### CODE REVIEW

File: refinery/core/views.py [hunk 1]
```diff
@@ -1094,7 +1095,12 @@ def delete(self, request, uuid):
                 content="User {} is not authenticated".format(request.user))
         else:
             try:
-                dataset_deleted = DataSet.objects.get(uuid=uuid).delete()
+                with transaction.atomic():
```
Comment (hackdna) [inline refinery/core/views.py]: There should be a comment explaining why this is necessary

File: refinery/core/views.py [hunk 2]
```diff
@@ -1094,7 +1095,12 @@ def delete(self, request, uuid):
...
+                    dataset_deleted = DataSet.objects.get(uuid=uuid).delete()
```
Comment (hackdna) [inline refinery/core/views.py]: This should be split up into get and delete for readability and to handle `DataSet.DoesNotExist` exception.

File: refinery/core/views.py [hunk 3]
```diff
@@ -1144,7 +1150,12 @@ def delete(self, request, uuid):
                 content="User {} is not authenticated".format(request.user))
         else:
             try:
-                analysis_deleted = Analysis.objects.get(uuid=uuid).delete()
+                with transaction.atomic():
```
Comment (hackdna) [inline refinery/core/views.py]: Same as above for DataSet

File: refinery/core/views.py (hunk 2 above)
Comment (scottx611x) [inline refinery/core/views.py]: Doubdle check DataSet.delete() may be handling exceptions there already

File: refinery/core/views.py (hunk 3 above)
Comment (scottx611x) [inline refinery/core/views.py]: double check Analysis.delete() as well

**Comment by scottx611x** [comment]
Transaction handling should really happen in Dataset.delete() & Analysis.delete()

**Comment by hackdna** [comment]
also "Avoid catching exceptions inside atomic!" https://docs.djangoproject.com/en/1.7/topics/db/transactions/

**Comment by mccalluc** [comment]
@scottx611x : This is pretty old: is it still active?

**Comment by scottx611x** [comment]
@mccalluc  Nope haven't touched this in awhile, the transaction handling needed here is much more involved than the simple fix in this PR

**Comment by scottx611x** [comment]
Closing. This is really outdated.

**Pull request closed without merging**